tools/release/sync-gutenberg-packages.js

```diff
@@ -120,8 +120,8 @@ function getMismatchedNonWordPressDependencies() {
 	;
 
 	// Ensure that all the conflicts can be resolved with the same version
-	const unresolvableConflicts = Object.entries( groupBy( versionConflicts, ( [name] ) => name ) )
-		.map( ( [name, group] ) => [name, group.map( ( [, { required }] ) => required )] )
+	const unresolvableConflicts = Object.entries( groupBy( versionConflicts, ( {name} ) => name ) )
+		.map( ( [name, group] ) => [name, uniq( group.map( ( { required } ) => required ) )] )
 		.filter( ( [, group] ) => group.length > 1 );
```

Comment:
this has gotten to the point where things are obviously confusing. could we improve this with more traditional iteration and type annotations?

```js
/**
 * @typedef DependencyRequirement
 * @property {string} name     Dependency name.
 * @property {string} required Minimum required version of dependency.
 */

/** @var {DependencyRequirement[]} */
const unresolvableConflicts = [];

/** @var {Record<string, DependencyRequirement[]>} */
const conflictGroups = groupBy( versionConflicts, version => version.name );

for ( const groupName in conflictGroups ) {
	const group = conflictGroups[ groupName ];

	if ( group.length <= 1 ) {
		continue;
	}

	unresolvableConflicts.push( [ groupName, uniq( group.map( version => version.required ] );
}
```

something in this direction?